src/Microsoft.AspNetCore.Blazor/Components/IHandleEvent.cs

```diff
@@ -15,6 +16,6 @@ public interface IHandleEvent
         /// </summary>
         /// <param name="handler">The event handler.</param>
         /// <param name="args">Arguments for the event handler.</param>
-        void HandleEvent(UIEventHandler handler, UIEventArgs args);
+        Task HandleEvent(Func<UIEventArgs, Task> handler, UIEventArgs args);
```

Comment:
I think we should consider dropping the nominal types for delegates and using `Action<T>` and `Func<T, Task>`. I think we'll end up with 6-7 event types minimum which means that we would be defining about 12-14 nominally-typed delegates.